prog/fsck_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package prog_test
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	. "github.com/google/syzkaller/prog"
+	"github.com/google/syzkaller/sys/targets"
+)
+
+// To get maximum test coverage here, install the following Debian packages:
+// dosfstools e2fsprogs btrfs-progs util-linux f2fs-tools jfsutils util-linux
+// dosfstools ocfs2-tools reiserfsprogs xfsprogs erofs-utils exfatprogs
+// gfs2-utils.
+
+const corruptedFs = "IAmACorruptedFs"
+
+func fsckAvailable(cmd string) bool {
+	_, err := exec.LookPath(strings.Fields(cmd)[0])
+	return err == nil
+}
+
+func TestFsck(t *testing.T) {
+	target, err := GetTarget(targets.Linux, targets.AMD64)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Use the images generated by syz-imagegen as a collection of clean file systems.
+	cleanFsProgs, err := filepath.Glob(filepath.Join("..", "sys", "linux", "test", "syz_mount_image_*_0"))
+	if err != nil {
+		t.Fatalf("directory read failed: %v", err)
+	}
+
+	for _, file := range cleanFsProgs {
+		sourceProg, err := os.ReadFile(file)
+		if err != nil {
+			t.Fatal(err)
+		}
+		p, err := target.Deserialize(sourceProg, NonStrict)
+		if err != nil {
+			t.Fatalf("failed to deserialize %s: %s", file, err)
+		}
+		p.ForEachAsset(func(name string, typ AssetType, r io.Reader, c *Call) {
+			if c.Meta.Attrs.Fsck == "" {
```

Comment:
There should also be some small test that checks whether `c.Meta.Attrs.Fsck` is indeed set to the value from the descriptions.

We have special `sys/test/` descriptions for that. You can define some syscall that sets  the `fsck` attribute and then, in the test, just check the value in `target.SyscallMap["your-syscall-name"]`.